Lib/importlib/_bootstrap.py

```diff
@@ -956,9 +956,19 @@ def _find_and_load_unlocked(name, import_):
 
 
 def _find_and_load(name, import_):
-    """Find and load the module, and release the import lock."""
-    with _ModuleLockManager(name):
-        return _find_and_load_unlocked(name, import_)
+    """Find and load the module."""
+    _imp.acquire_lock()
```

Comment:
This consolidation seems a little confusing to me, but I like the idea of simplifying the C code by moving the `ModuleNotFoundError` generation into Python.

How about adding a dedicated `_raise_for_halted_import(name, import_):` function that just handles the `module is None` case?